src/plugins/chessevent/data/chessevent_field_reader.py

```diff
@@ -0,0 +1,34 @@
+from typing import Type, Union, Optional, cast
+from enum import IntEnum
+
+class ChessEventFieldReader:
+    def __init__(self, data: dict[str, Union[bool, str, int, float, dict[int, float], None]]):
```

Comment:
I don't know about using `Union`, but the suggestion below doesn't feel right either :/
```suggestion
    def __init__(self, data: dict[str, bool | str | int | float | dict[int, float] | None]):
```